src/pip/_internal/vcs/git.py

```diff
@@ -163,16 +163,21 @@ def obtain(self, dest):
 
     def get_url(self, location):
         """Return URL of the first remote encountered."""
-        remotes = self.run_command(
-            ['config', '--get-regexp', r'remote\..*\.url'],
-            show_stdout=False, cwd=location)
-        remotes = remotes.splitlines()
-        found_remote = remotes[0]
-        for remote in remotes:
-            if remote.startswith('remote.origin.url '):
-                found_remote = remote
-                break
-        url = found_remote.split(' ')[1]
+        try:
+            remotes = self.run_command(
+                ['config', '--get-regexp', r'remote\..*\.url'],
+                show_stdout=False, cwd=location)
+            remotes = remotes.splitlines()
+            found_remote = remotes[0]
+            for remote in remotes:
+                if remote.startswith('remote.origin.url '):
+                    found_remote = remote
+                    break
+            url = found_remote.split(' ')[1]
+        except InstallationError:
+            # If the above command fails we have no git remotes. Just use the
+            # local repo path in that case instead.
+            url = 'file://' + location
```

Comment:
Does this need to be escaped? I would check elsewhere in the code for how this is done. There seem to be helper functions around this.

Reply:
We pass location to cwd as is without any escaping. If its good enough for the subprocess execution I'm hoping it is good enough for this string, but I may be missing something important.